### PR TITLE
doc: nits and fixes for thread API

### PR DESCRIPTION
**Pull request by tshepang**

### CODE REVIEW

File: src/libstd/thread.rs [hunk 1]
```diff
@@ -28,25 +28,25 @@
...
 //! When the main thread of a Rust program terminates, the entire program shuts
 //! down, even if other threads are still running. However, this module provides
 //! convenient facilities for automatically waiting for the termination of a
-//! child thread (i.e., join), described below.
+//! child thread (i.e., join).
 //!
 //! ## The `Thread` type
 //!
-//! Already-running threads are represented via the `Thread` type, which you can
+//! Threads are represented via the `Thread` type, which you can
```
Comment (huonw) [inline src/libstd/thread.rs]: `Thread` still only represents already-running threads.

File: src/libstd/thread.rs [hunk 2]
```diff
@@ -92,7 +90,7 @@
 //! terminates) when it is dropped. You can join the child thread in
 //! advance by calling the `join` method on the guard, which will also
 //! return the result produced by the thread.  A handle to the thread
-//! itself is available via the `thread` method on the join guard.
+//! itself is available via the `thread` method of the join guard.
 //!
 //! (Note: eventually, the `scoped` constructor will allow the parent and child
```
Comment (huonw) [inline src/libstd/thread.rs]: This paragraph can be deleted now, if you feel like it.

File: src/libstd/thread.rs (hunk 1 above)
Comment (tshepang) [inline src/libstd/thread.rs]: I removed *Already-running* because it doesn't work well with the line that follows it:

> By spawning a new thread, e.g. using the `thread::spawn` function.

File: src/libstd/thread.rs (hunk 2 above)
Comment (tshepang) [inline src/libstd/thread.rs]: Is that because the *eventually* part is now reality?

Comment (huonw) [inline src/libstd/thread.rs]: Yes.

Comment (tshepang) [inline src/libstd/thread.rs]: done, thanks

File: src/libstd/thread.rs [hunk 3]
```diff
@@ -28,25 +28,25 @@
 //! a thread will unwind the stack, running destructors and freeing
 //! owned resources. Thread panic is unrecoverable from within
 //! the panicking thread (i.e. there is no 'try/catch' in Rust), but
-//! panic may optionally be detected from a different thread. If
-//! the main thread panics the application will exit with a non-zero
+//! the panic may optionally be detected from a different thread. If
+//! the main thread panics, the application will exit with a non-zero
```
Comment (alexcrichton) [inline src/libstd/thread.rs]: I don't think this comma is necessary

File: src/libstd/thread.rs [hunk 4]
```diff
@@ -28,25 +28,25 @@
...
 //! exit code.
 //!
 //! When the main thread of a Rust program terminates, the entire program shuts
 //! down, even if other threads are still running. However, this module provides
 //! convenient facilities for automatically waiting for the termination of a
-//! child thread (i.e., join), described below.
```
Comment (alexcrichton) [inline src/libstd/thread.rs]: Why lose this?

File: src/libstd/thread.rs [hunk 5]
```diff
@@ -59,29 +59,27 @@
...
-//!     println!("Hello, World!");
-//!     // some computation here
+//!     // some work here
 //! });
 //! ```
 //!
 //! In this example, the spawned thread is "detached" from the current
-//! thread, meaning that it can outlive the thread that spawned
-//! it. (Note, however, that when the main thread terminates all
-//! detached threads are terminated as well.)
```
Comment (alexcrichton) [inline src/libstd/thread.rs]: How come this was re-worded?

File: src/libstd/thread.rs [hunk 6]
```diff
@@ -92,11 +90,7 @@
 //! terminates) when it is dropped. You can join the child thread in
 //! advance by calling the `join` method on the guard, which will also
 //! return the result produced by the thread.  A handle to the thread
-//! itself is available via the `thread` method on the join guard.
-//!
-//! (Note: eventually, the `scoped` constructor will allow the parent and child
-//! threads to data that lives on the parent thread's stack, but some language
-//! changes are needed before this is possible.)
+//! itself is available via the `thread` method of the join guard.
```
Comment (alexcrichton) [inline src/libstd/thread.rs]: Why did `on` switch to `of`?

File: src/libstd/thread.rs [hunk 7]
```diff
@@ -143,7 +137,7 @@
 //! * It avoids the need to allocate mutexes and condvars when building new
 //!   synchronization primitives; the threads already provide basic blocking/signaling.
 //!
-//! * It can be implemented highly efficiently on many platforms.
+//! * It can be implemented very efficiently on many platforms.
```
Comment (alexcrichton) [inline src/libstd/thread.rs]: Why this change?

File: src/libstd/thread.rs (hunk 3 above)
Comment (tshepang) [inline src/libstd/thread.rs]: If the comma is removed, the sentence will have to be slightly re-organised by starting with "the application...". Otherwise, the pause feels very needed.

File: src/libstd/thread.rs (hunk 7 above)
Comment (tshepang) [inline src/libstd/thread.rs]: it feels more natural: *very efficiently* vs. *highly efficiently*

File: src/libstd/thread.rs (hunk 4 above)
Comment (tshepang) [inline src/libstd/thread.rs]: Felt arbitrary and not needed. For consistency, we would need to mention *described below* for everything that will be described later.

File: src/libstd/thread.rs (hunk 5 above)
Comment (tshepang) [inline src/libstd/thread.rs]: - Having a whole sentence in brackets looks ugly
- It's good to explain that the thread that spwans another is the parent, (even though it may be obvious)

File: src/libstd/thread.rs (hunk 6 above)
Comment (tshepang) [inline src/libstd/thread.rs]: "on" feels less natural: *thread method on the join guard*. Is that common usage?

File: src/libstd/thread.rs (hunk 3 above)
Comment (jxcl) [inline src/libstd/thread.rs]: @alexcrichton The comma usage here is correct. See https://owl.english.purdue.edu/owl/resource/607/02/ item 2.

**Comment by alexcrichton** [comment]
r? @steveklabnik 

I may not be the best wordsmith

File: src/libstd/thread.rs (hunk 3 above)
Comment (steveklabnik) [inline src/libstd/thread.rs]: Some people leave it in, some leave it out. I generally put it in. It's fine either way.

File: src/libstd/thread.rs (hunk 6 above)
Comment (steveklabnik) [inline src/libstd/thread.rs]: Both are, it's a style thing.

File: src/libstd/thread.rs (hunk 7 above)
Comment (steveklabnik) [inline src/libstd/thread.rs]: "hightly efficiently" sounds a bit weird, I do like this better.

**Comment by tshepang** [comment]
wow, that felt long; thanks @steveklabnik
